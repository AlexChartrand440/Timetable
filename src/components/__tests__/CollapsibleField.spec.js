// @flow
import React from 'react';
import { Text, TouchableWithoutFeedback } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Collapsible from 'react-native-collapsible';
import CollapsibleField from '../CollapsibleField';

const props = {
  headerText: 'header',
  placeholder: 'header placeholer',
  content: <Text>only visible if not collapsed</Text>,
  meta: {},
};

describe('CollapsibleField component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CollapsibleField {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('initially is collapsed and toggles collapsed state when the header is pressed', () => {
    const wrapper = shallow(<CollapsibleField {...props} />);
    const button = wrapper.find(TouchableWithoutFeedback);

    expect(wrapper.find(Collapsible).prop('collapsed')).toBe(true);

    button.simulate('press');
    expect(wrapper.find(Collapsible).prop('collapsed')).toBe(false);

    button.simulate('press');
    expect(wrapper.find(Collapsible).prop('collapsed')).toBe(true);
  });
});
