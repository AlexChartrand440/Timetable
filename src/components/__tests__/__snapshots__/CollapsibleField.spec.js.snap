// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleField component renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <View>
      <View
        style={
          Object {
            "height": 48,
          }
        }
      >
        <TextInput
          allowFontScaling={undefined}
          editable={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="header placeholer"
          secureTextEntry={undefined}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "marginTop": 0,
                "paddingBottom": 2,
                "paddingTop": 2,
              },
              Object {
                "color": "rgb(117, 117, 117)",
                "fontSize": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {
                "flex": 1,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value="header"
        />
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 1,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#247BA0",
                "height": 1,
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        only visible if not collapsed
      </Text>
    </View>
  </View>
</View>
`;
